pkg/caveats/eval.go

```diff
@@ -51,8 +47,8 @@ func (cr CaveatResult) PartialValue() (*CompiledCaveat, error) {
 		return nil, fmt.Errorf("result is fully evaluated")
 	}
 
-	expr := interpreter.PruneAst(cr.parentCaveat.ast.Expr(), cr.details.State())
-	return &CompiledCaveat{cr.parentCaveat.celEnv, cel.ParsedExprToAst(&exprpb.ParsedExpr{Expr: expr}), cr.parentCaveat.name}, nil
+	expr := interpreter.PruneAst(cr.parentCaveat.ast.Expr(), cr.parentCaveat.ast.SourceInfo().GetMacroCalls(), cr.details.State())
```

Comment:
I don't know what this does. Does this further remove parts of the AST that aren't necessary for evaluation and getting debug symbols? Macros can and have been used in caveats, would that make troubleshooting errors more complicated if source info is not present?

Reply:
It prune the AST of any executed branches, simplifying it down to only the remaining branches. This is tested in the unit tests